docusaurus/docs-internals/editor-controls/data-controls/fileUpload.md

```diff
@@ -0,0 +1,177 @@
+
+### Return value
+
+The return value is string that, consist from `id` and `name` of the file.
+```js
+{
+  value: string;
+}
+```
+
+Example of value:
+```js
+{
+  value: "a8e41e74c4cba914345096f152500e9c894bd54607e553c658eae2c61c13829f|||eagle.jpg"
+}
+```
+
+### Usage
+
+
+#### Basic example
+Standard definition with only the required keys. This control will be displayed on all devices.
+
+```js
+{
+  id: "file",
+  type: "fileUpload",
+}
+```
+
+#### Label example
+Adding a label on the left side of the control.
+
+```js
+{
+  id: "file",
+  type: "fileUpload",
+  label: "File"
+}
+```
+
+#### Devices examples
+It will be rendered on all devices. This value can be skipped because it is set to `"all"` by default.
+
+```js
+{
+  id: "file",
+  type: "fileUpload",
+  devices: "all"
+}
+```
+
+Rendering will occur only on `desktop`.
+
+```js
+{
+  id: "file", 
+  type: "fileUpload",
+  devices: "desktop"
+}
+```
+The display is limited to responsive modes, specifically `tablet` and `mobile`.
+
+```js
+{
+  id: "file", 
+  type: "fileUpload",
+  devices: "responsive"
+}
+```
+
+#### Disabled examples
+
+Control will be disabled. Normally, here should be your dynamic condition.
+
+```js
+{
+  id: "file", 
+  type: "fileUpload", 
+  disabled: true
+}
+```
+
+Control will be disabled when `selectType` variable will be `"youtube"`.
+`getValue` is a getter function that allows us to retrieve the value of controls by their id.
+`"selectType"` is the id of the `"select"` control below.
+
+```js
+const getToolbarContols = ({ getValue }) => {
+  const selectType = getValue("selectType");
+
+  return [
+    {
+      id: "selectType",
+      type: "select",
+      choices: [
+        { title: "Youtube", value: "youtube" },
+        { title: "Custom", value: "custom" }
+      ]
+    },
+    {
+      id: "file",
+      type: "fileUpload",
+      disabled: videoType === "custom"
+    }
+  ]
+}
+```
+
+#### Helper examples
+
+The helper object contains a content property with the value `"Helper"`, which will be displayed as additional guidance or information for the user.
+
+```js
+{
+  id: "file",
+  type: "fileUpload",
+  helper: {
+    content: "Helper"
+  }
+}
+```
+
+When the helper object contains a position property with the value `"top-start"`, it indicates that the helper text will be displayed at the top left corner of the icon.
+
+```js
+{
+  id: "file",
+  type: "fileUpload",
+  helper: {
+    content: "Helper",
+    position: "top-start"
+  }
+}
+```
+
+#### `config?.allowedExtensions` example
+
+In this case `allowedExtensions` allows to upload only `.png`, `.jpg` and `.jpeg`, `.mp4` files.
```

Comment:
In this case `fileUpload` control allows to upload only `.png`, `.jpg`, `.jpeg` and `.mp4` files.

P.S. check if is grammatical correct